### PR TITLE
Doc: Move/rename 3 topics into app plugins how-to category

### LINKED ISSUE #925: Reorganize plugin-tools site based on new design plan
Fix problems with less than ideal plugin-tools organization. Incorporate feedback received in Q1 from many stakeholders, external and internal to the plugins team and approved in a design plan.

This process includes the following:
- Move or rename folders and files
- Fix broken links on site caused by move/renames
- Fix broken links on other sites, principally grafana/grafana and plugin-examples
- Create new topic placeholders for new topics, make edits and/or assign to developers to make edits
- Test all changes through build process on Docker site and dev site
```[tasklist]
### Tasks
- [x] Reorganize 'reference' section
- [x] Reorganize 'introduction' section
- [ ] Reorganize 'create-a-plugin/develop-a-plugin' section
- [ ] Reorganize 'create-a-plugin/extend-a-plugin' section
```

### PR DESCRIPTION
**Pull request by josmperez**
Complete 4th stage of the plugin-tools site reorganization project. Move/rename site, fix broken links.

Instructions for testing redirects: docusaurus/website/README.md#Redirects

Fixes: https://github.com/grafana/plugin-tools/issues/925

Testing: Refer to [design doc](https://docs.google.com/document/d/1T3v41Ik5hOYNL0xMtPLOiK38IPGUlvsEgzFkDKvsROM/edit)

### CODE REVIEW

File: docusaurus/docs/how-to-guides/ui-extensions/_category_.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "position": 10,
+  "position": 20,
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/ui-extensions/_category_.json:2]:
```suggestion
  "position": 40,
```

File: docusaurus/website/docusaurus.config.base.js [hunk 2]
```diff
@@ -113,6 +113,14 @@ const plugins = [
           from: ['/ui-extensions/create-an-extension-point'],
           to: '/how-to-guides/ui-extensions/create-an-extension-point',
         },
+        {
+          from: ['/create-a-plugin/extend-a-plugin/include-dashboards'],
+          to: '/how-to-guides/app-plugins/include-dashboards-in-app-plugins',
```
Comment (sympatheticmoose) [inline docusaurus/website/docusaurus.config.base.js:118]: I'd suggest removing this redirect
- its a fairly new topic so less impact
- we're splitting the content into multiple locations
- data sources are more common than apps, so the not-yet-created page would be a more appropriate redirect **IF** we need one

File: docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+---
+id: error-handling-in-app-plugins
+title: Error handling in app plugins
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md:3]: Not sure if we need _in app plugins_ in the sidebar? The additional context is provided by the folder and it just looks odd (to me at least). Could override with `sidebar_label: Error handling` and `sidebar_label: Include dashboards` for these panels

Screenshot 2024-07-01 at 11 30 16
Screenshot 2024-07-01 at 11 33 19

File: docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ keywords:
   - bundling
 ---
 
-# Include dashboards with a plugin
+# Include dashboards with an app plugin
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md:14]: below text still references data source plugins

File: docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md [hunk 5]
```diff
@@ -93,12 +80,6 @@ To facilitate user customization, create a dashboard data source variable. This
 
 To test your newly created dashboard, import the dashboard:
 
-### Import a dashboard into a data source plugin
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md:96]: @jackw does it make sense to keep Step 3 in its current state given app plugins automatically import?

The two suggestions I have are to change to "Step 3: Verify your dashboard imports successfully" or "Step 3: Add navigation links to your dashboard (optional)"

File: docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md [hunk 6]
```diff
@@ -11,7 +11,7 @@ keywords:
   - bundling
 ---
 
-# Include dashboards with a plugin
+# Include dashboards with an app plugin
 
 This guide explains how to add pre-configured dashboards into app and data source Grafana plugins. By integrating pre-configured dashboards into your plugin, you can provide your users with a ready-to-use template, freeing them from having to build a dashboard from scratch.
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md:16]:
```suggestion
This guide explains how to add pre-configured dashboards into app plugins. By integrating pre-configured dashboards into your plugin, you can provide your users with a ready-to-use template, freeing them from having to build a dashboard from scratch.
```

File: docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md [hunk 7]
```diff
@@ -21,19 +21,6 @@ We'll walk you through the process of bundling dashboards into plugins. The proc
 
 Start by creating the dashboard you want to bundle with your plugin. The [development environment](/get-started/set-up-development-environment) provided by create-plugin can aid with creating and testings dashboards. The following steps provide instructions for both data source and app plugins, including tips for crafting an effective dashboard.
```
Comment (sympatheticmoose) [inline docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md:22]:
```suggestion
Start by creating the dashboard you want to bundle with your plugin. The [development environment](/get-started/set-up-development-environment) provided by create-plugin can aid with creating and testing dashboards.
```

File: docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md (hunk 3 above)
Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md:3]: _in X plugin_ doesn't appear in the datasources PR by comparison

File: docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md (hunk 5 above)
Comment (jackw) [reply docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md:96]:
```
## Step 3: Import the dashboard into your plugin

### Import a dashboard into an app plugin

### (Optional) Add navigation links in an app plugin

```
I don't think it makes sense if this doc is now app specific. I would opt for "Step 3: Add navigation links to your dashboard (optional)" and probably add something above in step 2 that explains they're auto imported... e.g.

```
4. After adding the dashboard to your plugin, rebuild the plugin and restart Grafana to apply the new configuration.
5. App dashboards are automatically imported by the Grafana server. Navigate to your dashboard to verify it works as intended.
```
Does that make sense? WDYT?

Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md:96]: made these changes in https://github.com/grafana/plugin-tools/pull/978/commits/772378ccc59cac9a65ce16511cef3efff287f26b minor rewording and shifted (optional) to end of line as that made it fit on a single line and looked better.

File: docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md (hunk 3 above)
Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md:3]: I'm going to resolve this for now as it is the same with panels. We can revisit in the datasources PR or tidy up for consistency in a later PR

File: docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+```ts
+if (!numberField) {
+  throw new Error('Query result is missing a number field');
+}
+
+if (frame.length === 0) {
+  throw new Error('Query returned an empty result');
+}
+```
```
[the remark this answers is not in the document]
Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/app-plugins/error-handling-in-app-plugins.md]: resolved in https://github.com/grafana/plugin-tools/pull/978/commits/6fe7443b0db8037c1c789a75df780a26da8e6d85

File: docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md [hunk 9]
```diff
@@ -21,19 +21,6 @@ We'll walk you through the process of bundling dashboards into plugins. The proc
...
-In this step, we export the dashboard to a JSON file so it can be placed in a file along with your plugin source code:
-
-1. Open your dashboard within Grafana.
-1. Click the **Share** icon at the top-left of the dashboard.
-1. Click **Export**.
-1. Select **Export for sharing externally**, and then click **Save to file**.
-
-Exporting with this option replaces direct data source references with placeholders. This ensures the dashboard can make use of the user's data source instance when imported.
-
 ### Build a dashboard for an app plugin
```
[the remark this answers is not in the document]
Comment (sympatheticmoose) [reply docusaurus/docs/how-to-guides/app-plugins/include-dashboards-in-app-plugins.md]: resolved in https://github.com/grafana/plugin-tools/pull/978/commits/5c28b14545f17622f2d801adb81c300b6092572e

**Review by sympatheticmoose — APPROVED** [review]
@josmperez to help us get these wrapped up and not get in your way of the subsequent PRs, I've applied hopefully non-contentious changes and got to a point where I feel it could be merged. Please take a look at my changes and - if you are happy - go ahead and merge. Otherwise feel free to make adjustments/comments as appropriate.

**Review by academo — APPROVED** [review]

**Pull request merged**